pkg/attestation/attestation.go

```diff
@@ -0,0 +1,141 @@
+package attest
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"time"
+
+	"github.com/absmach/magistrala/pkg/errors"
+	"github.com/google/go-sev-guest/abi"
+	"github.com/google/go-sev-guest/proto/check"
+	"github.com/google/go-sev-guest/proto/sevsnp"
+	"github.com/google/go-sev-guest/validate"
+	"github.com/google/go-sev-guest/verify"
+	"github.com/google/go-sev-guest/verify/trust"
+	"github.com/google/logger"
+	"github.com/ultravioletrs/cocos/agent"
+	"github.com/ultravioletrs/cocos/agent/quoteprovider"
+)
+
+const (
+	cocosDirectory        = ".cocos"
+	caBundleName          = "ask_ark.pem"
+	attestationReportSize = 0x4A0
+)
+
+var (
+	AttConfigurationSEVSNP = AttestationConfiguration{}
+	timeout                = time.Minute * 2
+	maxTryDelay            = time.Second * 30
+)
+
+var (
+	errAttVerification = errors.New("attestation verification failed")
+	errAttValidation   = errors.New("attestation validation failed")
+)
+
+type AttestationConfiguration struct {
+	SNPPolicy   *check.Policy      `json:"snp_policy,omitempty"`
+	RootOfTrust *check.RootOfTrust `json:"root_of_trust,omitempty"`
+}
+
+func VerifyAttestationReportTLS(attestationBytes []byte, reportData []byte) int {
```

Comment:
the idiomatic golang way is to return an error

Reply:
Fixed. My mistake, this function was called by C at one point and that is why it had an int as a return type.